src/components/core/pivotTable/agGridDataSource.ts

```diff
@@ -126,6 +141,122 @@ export const getDataSourceRowsGetter = (
     };
 };
 
+export const executionToAGGridAdapter = (
+    executionResponses: Execution.IExecutionResponses,
+    resultSpec: AFM.IResultSpec = {},
+    intl: InjectedIntl,
+    options: IGridAdapterOptions = {},
+): IAgGridPage => {
+    const { makeRowGroups = false, addLoadingRenderer = null, columnDefOptions } = options;
+
+    const {
+        executionResponse: { dimensions },
+        executionResult: { data, headerItems, totals },
+    } = executionResponses;
+
+    const columnAttributeHeaderCount = dimensions[1].headers.filter(
+        (header: Execution.IHeader) => !!(header as Execution.IAttributeHeader).attributeHeader,
+    ).length;
+
+    const columnHeaders: IGridHeader[] = getColumnHeaders(
+        headerItems[1],
+        dimensions[1].headers,
+        columnDefOptions,
+    );
+    const groupColumnHeaders: IGridHeader[] =
+        columnAttributeHeaderCount > 0
+            ? [
+                  {
+                      headerName: dimensions[1].headers
+                          .filter(header => Execution.isAttributeHeader(header))
+                          .map((header: Execution.IAttributeHeader) => {
+                              return getMappingHeaderName(header);
+                          })
+                          .filter((item: string) => item !== null)
+                          .join(" › "),
```

Comment:
Detail: I know you did not change this, but this should also be in the `agGridConst.ts` file.

Reply:
create constant for this